File: processing/src/main/scala/com/datakata/flink/model/SalesEvent.scala
package com.datakata.flink.model

case class SalesEvent(
    saleId: String,
    salesmanName: String,
    city: String,
    country: String,
    amount: Double,
    product: String,
    eventTime: Long,
    source: String,
    ingestionTime: Long
)

File: processing/src/main/java/com/datakata/flink/model/SalesEvent.java
package com.datakata.flink.model;

public record SalesEvent(
    String saleId,
    String salesmanName,
    String city,
    String country,
    double amount,
    String product,
    long eventTime,
    String source,
    long ingestionTime
) {
    public SalesEvent withSource(String source) {
        return new SalesEvent(saleId, salesmanName, city, country, amount, product, eventTime, source, ingestionTime);
    }
}
